protocol/consistencychecks.go

```diff
@@ -278,27 +289,29 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 			if err := cc.verifyReturnedPromise(df, key); err != nil {
 				return err
 			}
-			cc.savePromise(uname, df.TB)
+			cc.TBs[uname] = df.TB
+			cc.RegEpoch[uname] = cc.SavedSTR.Epoch
 		}
 		return nil
 
 	case KeyLookupType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		ap := df.AP
-		str := df.STR
 		proofType := ap.ProofType()
 		switch {
-		case msg.Error == ReqSuccess && proofType == m.ProofOfInclusion:
-			if err := cc.verifyFulfilledPromise(uname, str, ap); err != nil {
-				return err
-			}
-			delete(cc.TBs, uname)
+		// FIXME: We don' support this for now,
+		// until we know how to cache the issued epoch. (See #116)
+		// case msg.Error == ReqSuccess && proofType == m.ProofOfInclusion:
+		// 	if err := cc.verifyFulfilledPromise(uname, str, ap); err != nil {
+		// 		return err
+		// 	}
+		// 	delete(cc.TBs, uname)
 
 		case msg.Error == ReqSuccess && proofType == m.ProofOfAbsence:
 			if err := cc.verifyReturnedPromise(df, key); err != nil {
 				return err
 			}
-			cc.savePromise(uname, df.TB)
+			cc.TBs[uname] = df.TB
```

Comment:
Nevermind, I understand now. I was a bit confused by the FIXME comment, since it seemed to me that we were caching the issued epoch via the `RegEpoch` map. But I think it's best to address #116 in a separate PR.